emailCrawler.java

```diff
@@ -49,16 +41,14 @@ public static void main(String args[]) throws IOException {
 	public static void printEmail(ArrayList<String> urlList) throws IOException {
 		for (String url : urlList) {
 			Document doc = Jsoup.connect(url).get();
-
 			// email format
 			Matcher m = Pattern.compile("[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\\.[a-zA-Z0-9-.]+")
 					.matcher(doc.select("a").toString());
 			while (m.find()) {
 				String email = m.group();
 				if (!emailList.contains(email)) {
```

Comment:
Since emailList is a HashSet, this check is not necessary.  The only purpose seems to be to make sure duplicate emails are not printed more than once.
A better way would be to collect the email for all of the pages in the urlList first then iterating the completed emailList set in the end and printing each item.